creme/billing/views/line.py

```diff
@@ -117,6 +118,25 @@ def get_item_class(self):
         return cls
 
 
+class ReorderLines(ReorderInstances):
+    pk_ct_kwarg: str = 'ctype_id'
+    pk_url_kwarg: str = 'line_id'
+    target_order_post_argument: str = 'target'
+    order_field_name: str = 'order'
+    lines_registry = lines_registry
+
+    def get_line_model(self):
+        model = get_ctype_or_404(self.kwargs[self.pk_ct_kwarg]).model_class()
+
+        if model not in self.lines_registry:
+            raise ConflictError('This entity is not a billing line')
+
+        return model
+
+    def get_queryset(self):
+        return self.get_line_model().objects.all()
+
```

Comment:
Hum there are 2 big issues:
 - There is no permission checking.
 - You retrieve all the lines of a type (instead of retrieving only  the lines related to an Invoice/Quote/...)

I presume you should POST the Invoice/Quote/.. ID & the lines' ContentType ID. Then retrieve the main entity, check the edition permission, then retrieve the related lines.
( if you look at `creme.creme_config.views.generics_views.Reorder`, you'll see that the call `self.get_model_conf()` lead to permission checking `self.request.user.has_perm_to_admin_or_die(app_name)` in `AppRegistryMixin`).